stream_alert/alert_processor/outputs/output_base.py

```diff
@@ -243,12 +250,90 @@ def _kms_decrypt(self, data):
             str: Decrypted json string
         """
         try:
-            client = boto3.client('kms', region_name=self.region)
+            client = boto3.client('kms', region_name=self._region)
             response = client.decrypt(CiphertextBlob=data)
             return response['Plaintext']
         except ClientError as err:
             LOGGER.error('an error occurred during credentials decryption: %s', err.response)
 
+    @staticmethod
+    def get_formatted_output_credentials_name(service_name, descriptor):
+        """Formats the output name for this credential by combining the service
+        and the descriptor.
+
+        Args:
+            service_name (str): Service name on output class (i.e. "pagerduty", "demisto")
+            descriptor (str): Service destination (ie: slack channel, pd integration)
+
+        Returns:
+            str: Formatted credential name (ie: slack_ryandchannel)
+        """
+        cred_name = str(service_name)
+
+        # should descriptor be enforced in all rules?
```

Comment:
we should probably have a "validator" for the keyword arguments passed to the `@rule` decorator that would 'lint' or enforce this, but we don't yet, so please leave this :)

Reply:
I just filed an issue for this... #879